interfaces/apparmor/spec.go

```diff
@@ -42,7 +42,7 @@ type Specification struct {
 	snippets map[string][]string
 	// updateNS describe parts of apparmor policy for snap-update-ns executing
 	// on behalf of a given snap.
-	updateNS []string
+	updateNS strutil.OrderedSet
```

Comment:
I not happy that we loose the input information. The input snippet is silently dropped in the input, what may cause the output to be confusing. Eg:

```
// snippet 1
spec.AddUpdateNs("/foo rwx\n")
// snippet 2
spec.AddUpdateNS("# the rule below does magic, needed for you-name-it-here\n")
spec.AddUpdateNs("/foo rwx\n")
// snippet 3
spec.AddUpdateNs("/bar rxi\n")
```
If I understand correctly what will happen, the output will look like this:
```
/foo rwx
# the rule below does magic, needed for you-name-it-here\
/bar rxi
```

I think it'd be better to preserve the input snippet but have it commented out, or leave a trace that it's been deduplicated, eg.
```
/foo rwx
# the rule below does magic, needed for you-name-it-here
# (deduplicated block start, prevoious in line 123):
# /foo rwx
# (deduplicated block end)
/bar rxi
```

I do hope that AppArmor parser does not choke on comment lines though.

Reply:
It doesn't choke on comments but it was NACKed by Jamie as far as I recall. He asked not to do this.

Reply:
As discussed that on IRC, we're aiming for shorter profile files.

Reply:
So, to be clear, the affected profiles in question ended up being 17500+ lines long, taking up 1M each. This takes up space which we don't need to use, but more importantly, and while I appreciate the sentiment, a 17500 line profile, commented or not, is pretty unreadable for auditing. It seems the least worst option is a smaller profile for this *auto-generated* snap-update-ns profile. For the app templates and snippets that are handcrafted, we expressly want to keep everything.